### PR TITLE
add iControlLX IPAM infoblox worker

### PR DESCRIPTION
**Pull request by nmenant**

### CODE REVIEW

File: icontrollx/ipam-infoblox/nodejs/infoblox-extension.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+* this iControl LX extension can be used to communicate with infoblox
+* DDI solution. It is able to request an IP address from a specific subnet/zone
+* You'll need to make sure to setup your subnet and related zone accordingly
+* in Infoblox. When you request an IP address, you need to specify the FQDN
+* related to the service that will leverage this IP. The domain name related to
+* the FQDN specified must exist in Infoblox as a zone
+*/
+
+var logger = require('f5-logger').getInstance();
+var promise = require("/usr/share/rest/node/node_modules/promise");
```
Comment (elv1s) [inline icontrollx/ipam-infoblox/nodejs/infoblox-extension.js:11]: This can be removed. You are using the native `Promise`

**Review by elv1s — COMMENTED** [review]
1 minor comment, otherwise looks good!

File: icontrollx/ipam-infoblox/nodejs/infoblox-extension.js (hunk 1 above)
Comment (nmenant) [reply icontrollx/ipam-infoblox/nodejs/infoblox-extension.js:11]: i think on iWF i had some issues without it. Let me check tomorrow.

File: icontrollx/ipam-infoblox/utils/infoblox-utils.js [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+  /*
+  * Here we provide the information to communicate with infoblox:
+  *   - Infoblox IP
+  *   - Infoblox Login / Password
+  *   - Infloblox subnet
+  */
+
+  var infobloxIP = "10.100.60.71";
+  var infobloxLogin = "admin";
+  var infobloxPassword = "51L1c0mp"
```
Comment (atenenb) [inline icontrollx/ipam-infoblox/utils/infoblox-utils.js:28]: Are you sure you want to put credentials in the source code?

Comment (nmenant) [reply icontrollx/ipam-infoblox/utils/infoblox-utils.js:28]: right now we don't have a way to store secrets so don't have a choice. Those credentials are related to a private env but you're right that i should at least put something else here

**Comment by nmenant** [comment]
ok i'm playing with AS3 and BIG-IPv13 and seems like we changed a few things like you need to use node_modules in your rpm (for example for request). I'll update the code to fix this.

**Pull request closed without merging**